lib/elixir/lib/kernel.ex

```diff
@@ -3271,7 +3271,12 @@ defmodule Kernel do
       if Map.has_key?(fields, :message) do
         @spec message(Exception.t) :: String.t
         def message(exception) do
-          exception.message
+          cond do
+            is_nil(exception.message) ->
```

Comment:
@eksperimental this should already be handled by `Exception.message/1` which is the default (and only) mechanism  for reliable getting error messages from exceptions in Elixir.